client/proposalmsgs_test.go

```diff
@@ -135,6 +141,23 @@ func TestSubChannelProposalReqProposalID(t *testing.T) {
 	assert.NotEqual(t, s, c3.ProposalID())
 }
 
+func TestVirtualChannelProposalReqProposalID(t *testing.T) {
+	rng := pkgtest.Prng(t)
+
+	prop, err := clienttest.NewRandomVirtualChannelProposal(rng)
+	require.NoError(t, err)
+
+	testProp := *prop
+	assert.Equal(t, prop.ProposalID(), testProp.ProposalID())
```

Comment:
Just to clarify:  
You use the embedded `BaseChannelProposal.ProposalID` here since `VirtualChannelProposal` does not have its own `ProposalID` function.  
Multiple different `VirtualChannelProposal`s *could* now have the same proposal ID, if the nonce is not unique.  

I am just thinking about if this is a problem or not, it seems weird but i think it is fine.  
Do you see a problem with it @sebastianst ?

Reply:
I am doing that since it is unique do to the nonce. No need to encode the whole set of parameters.

Reply:
AFAIR, we don't use the proposal id in the protocol for security reasons, right? I mean, it's not a commitment to all proposal params like the channel id is a commitment to the channel params. We rather use it as a means to have a reference id in the proposal protocol. If that is the case, I think it's fine like this. If one participant changes some params, it shouldn't (hopefully) be a security concern because only what is eventually signed (first state) matters.

But maybe good to double check in how far we rely on the proposal id to be a commitment to the proposal parameters.

Reply:
Yes, that is my understanding as well.